### PR TITLE
Add missing doc comments for connection manager module

### PR DESCRIPTION
**Pull request by RyanLassigBanks**
Adds missing doc comments for connection manager and pacemaker modules.

Signed-off-by: Ryan Banks <rbanks@bitwise.io>

### CODE REVIEW

File: libsplinter/src/network/connection_manager/mod.rs [hunk 1]
```diff
@@ -673,6 +740,16 @@ where
         }
     }
 
+    /// Adds outbound connection to matrix life cycle after it has been authorized
```
Comment (peterschwarz) [inline libsplinter/src/network/connection_manager/mod.rs:743]: Mention that these connections are not eligible for re-connection attempts.

File: libsplinter/src/network/connection_manager/pacemaker.rs [hunk 2]
```diff
@@ -22,6 +22,11 @@ use std::time::Duration;
 
 use crate::network::connection_manager::error::ConnectionManagerError;
 
+/// Pacemaker is responsible for periodically sending a message to
+/// the connection manager that instructs it to send a heartbeat
```
Comment (peterschwarz) [inline libsplinter/src/network/connection_manager/pacemaker.rs:26]: This has actually been generalized to send a message to any sender on an interval.  This struct isn't responsible for any of the connection manager's behaviour on each clock tick.

Comment (RyanLassigBanks) [reply libsplinter/src/network/connection_manager/pacemaker.rs:26]: Does anything else other than ConnectionManager instantiate and make use of a Pacemaker?

Comment (peterschwarz) [reply libsplinter/src/network/connection_manager/pacemaker.rs:26]: Not yet.  But that doesn't mean the doc for this struct should describe connection manager internals.  Connection manager's documentation should describe that it does periodic heartbeats.

Comment (peterschwarz) [reply libsplinter/src/network/connection_manager/pacemaker.rs:26]: s/to the connection manager...heartbeat/over a channel.

Comment (RyanLassigBanks) [reply libsplinter/src/network/connection_manager/pacemaker.rs:26]: Appended the phrase "over a channel"

Comment (RyanLassigBanks) [reply libsplinter/src/network/connection_manager/pacemaker.rs:26]: Appended phrase "over a channel"

File: libsplinter/src/network/connection_manager/mod.rs [hunk 3]
```diff
@@ -713,6 +770,12 @@ where
         }
     }
 
+    /// Adds inbound connection to matrix life cycle after it has been authorized
+    ///
+    /// # Errors
+    ///
+    /// Returns a connection manager error is the connection is unauthorized or
```
Comment (chenette) [inline libsplinter/src/network/connection_manager/mod.rs:777]: Typo: "is the connection" should be "if the connection"

File: libsplinter/src/network/connection_manager/mod.rs [hunk 4]
```diff
@@ -550,15 +575,15 @@ impl Connector {
...
+/// Sends a shutdown signal to the connection manager.
 #[derive(Clone)]
 pub struct ShutdownSignaler {
     sender: Sender<CmMessage>,
     pacemaker_shutdown_signaler: pacemaker::ShutdownSignaler,
 }
 
 impl ShutdownSignaler {
-    /// Signal the ConnectionManager to shutdown.
+    /// Signal the to shutdown.
```
Comment (vaporos) [inline libsplinter/src/network/connection_manager/mod.rs:586]: Looks like a typo

File: libsplinter/src/network/connection_manager/mod.rs [hunk 5]
```diff
@@ -607,6 +635,9 @@ enum ConnectionMetadataExt {
     },
 }
 
+/// Struct describing the connection manager's internal state and handling
+/// requests sent to the connection manager by it's Connectors. Connection state
```
Comment (chenette) [inline libsplinter/src/network/connection_manager/mod.rs:639]: Typo: change it's to its

File: libsplinter/src/network/connection_manager/mod.rs [hunk 6]
```diff
@@ -758,6 +791,17 @@ where
         }
     }
 
+    /// Adds outbound connection to matrix life cycle after it has been authorized
+    /// these connections are cannot be reconnected when dropped or lost.
```
Comment (chenette) [inline libsplinter/src/network/connection_manager/mod.rs:795]: "... after it has been authorized these connections" doesn't parse.  I can't tell if punctuation is missing or if words need to be added (or removed).  Also, "it" is unclear - grammatically, it refers to the matrix life cycle, but I don't think that's what you meant. 

Here's my guess at the wording and punctuation changes that are needed:

"Adds the(?) [[or an?]] outbound connection to the(?) matrix life cycle. After a connection has been authorized, it cannot be reconnected when dropped or lost."

File: libsplinter/src/network/connection_manager/mod.rs [hunk 7]
```diff
@@ -758,6 +791,17 @@ where
...
+    ///
+    /// # Returns
+    ///
+    /// A string representing the Connection ID.
+    ///
+    /// # Errors
+    ///
+    /// Returns a connection manager error is the connection is unauthorized or
+    /// is the life cycle fails to add the connection.
```
Comment (chenette) [inline libsplinter/src/network/connection_manager/mod.rs:804]: Two typos:  Change is to if:

"Returns a connection manager error is ..."
"... is the life cycle fails"

File: libsplinter/src/network/connection_manager/mod.rs [hunk 8]
```diff
@@ -817,6 +861,12 @@ where
         }
     }
 
+    /// Adds inbound connection to matrix life cycle after it has been authorized.
+    ///
+    /// # Errors
+    ///
+    /// Returns a connection manager error if the connection is unauthorized or
+    /// is the life cycle fails to add the connection.
```
Comment (chenette) [inline libsplinter/src/network/connection_manager/mod.rs:869]: Typo: s/is/if/

File: libsplinter/src/network/connection_manager/mod.rs [hunk 9]
```diff
@@ -872,6 +922,16 @@ where
...
 
+    /// Removes connection from state.
+    ///
+    /// # Returns
+    ///
+    /// Returns metadata for the connection if available.
+    ///
+    /// # Errors
+    ///
+    /// ConnectonManagerError if the connection cannot be removed from
```
Comment (chenette) [inline libsplinter/src/network/connection_manager/mod.rs:933]: Typo - the error name is missing an i.

File: libsplinter/src/network/connection_manager/mod.rs [hunk 10]
```diff
@@ -985,6 +1051,7 @@ where
     }
 }
 
+/// Auxiliary method for handling request send to connection manager.
```
Comment (chenette) [inline libsplinter/src/network/connection_manager/mod.rs:1054]: Doesn't parse - a word or words appear to be missing from "handling request send to". 

Also (as long as you're changing this line) there should be a "the" before "connection manager". This issue occurs in several other comments, too.

**Comment by peterschwarz** [comment]
The commit message still includes the pacemaker in its list of changes.

**Pull request merged**